### ISSUE TITLE
[Python] Implement new concept exercise: int-basic

### ISSUE THREAD

**Issue by aldraco** [exercism/v3#1025]
This issue describes how to implement the `int-basic` concept exercise for the Python track.

## Goal

The goal of this exercise is to teach the student how `int` is implemented and used in Python. We will teach the `int` type by showing the student how to use it.

## Learning objectives

 - create an `int` object, via `int` literal declaration
 - cast a string to an `int` using the builtin `int` function
 - perform arithmetic with `int` types
 - perform specifically integer division (as opposed to general division)
 - comparison operators (`>`, `<`, `==`)

## Out of scope

 - max size of `int` (technically unbounded; the "max value" will depend on the architecture it's run on)
 - working with `floats`. We should distinguish between this type and `int`, but `float` types are taught in another lesson.
 - covering every single mathematical operator (a few will do)
 - covering that `bool` is a subtype of `int` (i.e. `1` and `0`)
 - checking if a string is a number, i.e. `"string".isdigit()` or the like
 - catching/raising `Exceptions`

(The last 2 items imply that the solution should not worry about feeding in strings for the student to cast which are not numbers, i.e. they should not need to know how to catch Exceptions when trying to execute `int("notanumber")` and shouldn't be required to check the input, i.e. `input_string.isdigit()`.)

## Concepts

 - `int-basic`

## Prerequisites

 - `builtin_types`
 - `data-types`

## Resources to refer to

 - [Python builtin types: numeric docs](https://docs.python.org/3/library/stdtypes.html#typesnumeric)
 - [Python `int` function docs](https://docs.python.org/3/library/functions.html#int)

### Hints

Hints should link to the [Python docs about numeric types](https://docs.python.org/3/library/stdtypes.html#typesnumeric).

### After

Once the student has learned the basics of the `int` type, they can be introduced to `bool` as a subtype of `int`, represented as `0` and `1` (see below). 

```
>>> 1 == True
True
>>> 1 == False
False
```

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using `unittest.TestCase` and the test file named `ints_basic_test.py`.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by yawpitch** [comment]
@aldraco I agree that max bound is out of scope of `int-basic`, however it may be good to clarify the point about system architecture: **sys.maxint** limits the number of items that can appear in a collection, it does _not_ represent the largest possible **int** in Python 3: `s = (sys.maxsize ** 10000)` will happily let you work with an `int` that is 6,300,000 bits long on both a 32 bit or a 64 bit machine, assuming you've got at least that much memory available ... basically all `int` instances in Python 3 are bignum.

**Comment by aldraco** [comment]
(very late response) huh, thanks for the clarification @yawpitch! Didn't know that. 

I see that you edited some things for naming purposes - for future context, what were those changes? I'd like to know before writing another Issue and potentially naming things incorrectly. Thanks!

**Comment by yawpitch** [comment]
@aldraco yeah that little dropdown by the "edited" a the top of the comment box can be a lifesaver.

For the record, we renamed all the `[type]-basic` exercises to just `[type]s`, so as not to imply a hierarchy.

**Comment by SleeplessByte** [comment]
As we're not teaching _types_, perhaps you want to do `integers` and `fractionals`/`floating-points`?

**Comment by yawpitch** [comment]
> As we're not teaching _types_, perhaps you want to do `integers` (and another exercise `fractionals`/`floating-points`)?

Effectively Python has a 1-to-1 equivalency between all three of those concepts and the one type that exists to address them, so it's mainly a question of preferring long-form names to short ones.

**Comment by SleeplessByte** [comment]
Cool! Thanks for explaining :). In that case, I would go for the long form, as we're already using `floating-point-numbers` and not `floats`. The short-form is less accessible and _feels_ like a type. But but but don't mind me -- just a suggestion 😄
